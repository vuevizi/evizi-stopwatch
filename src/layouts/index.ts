import MainLayout from "./MainLayout/MainLayout";
import Header from "./Header/Header";
import Footer from "./Footer/Footer";
export {Header,MainLayout,Footer}