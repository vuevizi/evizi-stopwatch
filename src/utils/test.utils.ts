export const findTestElement = (component: any, attr: string) => {
  const findElement: any = component.find(`[data-test="${attr}"]`);
  return findElement;
};
