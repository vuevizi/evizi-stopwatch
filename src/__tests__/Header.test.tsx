import {Header} from '../layouts';
import {shallow} from 'enzyme';
import {findTestElement} from '../utils/test.utils';
import toJson from 'enzyme-to-json';

const setUpComponent = (props = {}) => {
  const component: any = shallow(<Header {...props} />);
  return component;
};
describe('Header', () => {
  let component: any;
  beforeEach(() => {
    component = setUpComponent();
  });
  test('renders without crash', () => {
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
  test('renders logo correctly', () => {
    const headerContainer = findTestElement(component, 'header-container');
    const headerLogo = findTestElement(headerContainer, 'header-logo');
    expect(headerLogo.length).toBe(1);
    expect(headerLogo.text()).toBe('Stop Watch');
  });
});
