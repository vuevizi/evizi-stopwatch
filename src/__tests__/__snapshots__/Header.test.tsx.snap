// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crash 1`] = `
<styled.header
  data-test="header-container"
>
  <styled.h1
    data-test="header-logo"
  >
    Stop Watch
  </styled.h1>
</styled.header>
`;
